### PR TITLE
Feature/network request

### LINKED ISSUE #13: Build out CardForm component
- [x] Add controlled form to component
- [x] Add handleChange methods to inputs
- [x] Add useState hook to component
- [x] Add fetch call to get "tidbit" for card
- [x] Add generate card button

### PR DESCRIPTION
**Pull request by camianderson**
Hello, team! Please review this PR!

Where should your reviewer start?
- Line number 16 in CardForm.

What (if anything) features are you implementing?
- Fetch Request to call the Compliment API;
- Created a card to pass to the state of the APP component.

What (if anything) did you refactor?
- Nothing!

Does this PR close any open issues?
- Closes #13 

Thank you!!

### CODE REVIEW

File: src/Components/App/App.tsx [hunk 1]
```diff
@@ -1,19 +1,24 @@
...
 import { Route } from "react-router-dom"
 import NavBar from '../NavBar/NavBar';
 import CardForm from '../CardForm/CardForm';
 import CardPreview from '../CardPreview/CardPreview';
 import SavedCards from '../SavedCards/SavedCards';
 
-
 const App: FC = () => {
 
+  const [savedCards, setSavedCards] = useState<{}[]>([])
```
Comment (GrantXBeard) [inline src/Components/App/App.tsx:11]: Great job here!

File: src/Components/CardForm/CardForm.tsx [hunk 2]
```diff
@@ -23,14 +51,30 @@ const CardForm: FC = () => {
...
             <textarea placeholder='Add message here...' name='message-input' onChange={ event => handleChange(event)} value={message} data-cy='message-input'/>
             <label htmlFor='from-input'>From: 
                 <input type='text' name='from-input' onChange={ event => handleChange(event)} value={from} data-cy='from-input'/> 
             </label>
+            <div>
+                <Link to="/preview-card">
+                    <button onClick={(event) => createCard(event)}>Make my card!</button>
+                </Link>
+                <button onClick={(event) => {
+                    event.preventDefault()
```
Comment (GrantXBeard) [inline src/Components/CardForm/CardForm.tsx:64]: Nice job adding the event.preventDefault() to the onClick

**Review by GrantXBeard — APPROVED** [review]
Great work team

File: src/Components/App/App.tsx (hunk 1 above)
Comment (sertmer) [inline src/Components/App/App.tsx:11]: ooooo gettin' Hooked, sick! Y'all wanna teach the hooks lesson for me 😆

File: src/Components/CardForm/CardForm.tsx [hunk 3]
```diff
@@ -1,13 +1,41 @@
-import React, { FC, useState, ChangeEvent } from 'react';
+import React, { FC, useState, ChangeEvent, useEffect } from 'react';
+import { Link } from 'react-router-dom';
 import './CardForm.css';
 
-const CardForm: FC = () => {
+interface Props {
+    addCard:(card: object) => void
+}
+
```
Comment (sertmer) [inline src/Components/CardForm/CardForm.tsx:8]: Lets use this as an opportunity to practice talking about our new tech - would some of you be willing to tell me what this code is doing?

File: src/Components/CardForm/CardForm.tsx [hunk 4]
```diff
@@ -1,13 +1,41 @@
...
     const [quote, setQuote] = useState<string>('')
     const [message, setMessage] = useState<string>('')
     const [from, setFrom] = useState<string>('')
 
+    const getCompliment = async (): Promise<any> => {
+        const URL = 'https://complimentr.com/api'
+        const response = await fetch(URL)
+        const quote = await response.json()
+        setQuote(quote.compliment)
+    }
```
Comment (sertmer) [inline src/Components/CardForm/CardForm.tsx:21]: Is it possible to break this out into an API calls file?

**Review by sertmer — COMMENTED** [review]
Thank you! Check those comments when you get a chance!

**Pull request merged**